src/index.js

```diff
@@ -3,10 +3,15 @@ var WriteData = require('./write-data');
 
 var main = function(options) {
   console.log("Gathering Manifest data...");
-  var resources = WalkManifest(options.decrypt, options.output, options.input);
-
-  console.log("Downloading additional data...");
-  return WriteData(options.decrypt, options.concurrency, resources);
+  var options = {decrypt: options.decrypt, basedir: options.output, uri: options.input};
+  WalkManifest(options)
+    .then(function(resources) {
+      console.log("Downloading additional data...");
+      return WriteData(options.decrypt, options.concurrency, resources);
+    })
+    .catch(function(err) {
```

Comment:
this isn't necessary.

Reply:
The `keyError` handler could likely also be removed and just allow the error to bubble